### PR TITLE
make destructor for FastTrackerRecHit virtual

### PR DESCRIPTION
**Pull request by lveldere**

### CODE REVIEW

**Comment by lveldere** [comment]
please test

File: DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ class FastTrackerRecHit : public BaseTrackerRecHit
     
     /// destructor
     ///
-    ~FastTrackerRecHit() {}
+    virtual ~FastTrackerRecHit() {}
```
Comment (VinInn) [inline DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h]: Not needed...
acutally 
```~FastTrackerRecHit()=default```
is the most correct declaration here...

Comment (lveldere) [inline DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h]: Hi Vincenzo

On the internet I read

"Deleting a derived class object using a pointer to a base class that has a non-virtual destructor results in undefined behavior."

Is that not true (anymore)?

Comment (makortel) [inline DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h]: Still true, but the base class `TrackingRecHit` already has a virtual destructor (https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/TrackingRecHit/interface/TrackingRecHit.h#L67). Essentially the "virtualness" of a destructor is inherited.

(nevertheless some coding styles suggest to explicitly declare the destructors of derived classes virtual to make it more clear that the class hierarchy is polymorphic)

Comment (cvuosalo) [inline DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h]: I agree with Vincenzo. We want to say that the default destructor is to be used here, not that we want to overload the base class destructor. Please change to:
```
~FastTrackerRecHit()=default;
```

**Comment by cvuosalo** [comment]
@lveldere: When might this PR get revised?

**Comment by slava77** [comment]
@lveldere please clarify on the status of this PR
- is the update mentioned in https://github.com/cms-sw/cmssw/pull/14177#discussion_r61178111 going to be implemented?
- is this still needed for 80X?

Thank you.

**Comment by slava77** [comment]
@lveldere 
ping

**Pull request closed without merging**